jre/javatests/com/google/j2cl/jre/java9/util/stream/DoubleStreamTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.j2cl.jre.java9.util.stream;
+
+import java.util.stream.DoubleStream;
+
+import com.google.j2cl.jre.java.util.EmulTestBase;
+
+/**
+ * Tests for java.util.stream.DoubleStream Java 9 API emulation.
+ */
+public class DoubleStreamTest extends EmulTestBase {
+
+  public void testTakeWhile() {
```

Comment:
Reading the javadoc, API contract looks complicated but the tests seems very light here and doesn't explain which contracts are tested. I think we need the beef these up some.